src/CommunityToolkit.Maui.Core/Essentials/SpeechToText/SpeechToTextImplementation.windows.cs

```diff
@@ -0,0 +1,158 @@
+using System.Globalization;
+using System.Speech.Recognition;
+using Microsoft.Maui.ApplicationModel;
+using Microsoft.Maui.Networking;
+using Windows.Globalization;
+using Windows.Media.Capture;
+using Windows.Media.SpeechRecognition;
+using SpeechRecognizer = Windows.Media.SpeechRecognition.SpeechRecognizer;
+
+namespace CommunityToolkit.Maui.Media;
+
+/// <inheritdoc />
+sealed class SpeechToTextImplementation : ISpeechToText
+{
+	const uint privacyStatementDeclinedCode = 0x80045509;
+	const int noCaptureDevicesCode = -1072845856;
+
+	string? recognitionText;
+	SpeechRecognizer? speechRecognizer;
+	SpeechRecognitionEngine? speechRecognitionEngine;
+
+	/// <inheritdoc />
+	public async ValueTask DisposeAsync()
+	{
+		await StopRecording();
+		StopOfflineRecording();
+		speechRecognitionEngine?.Dispose();
+		speechRecognizer?.Dispose();
+	}
+
+	/// <inheritdoc />
+	public async Task<string> ListenAsync(CultureInfo culture,
+		IProgress<string>? recognitionResult,
+		CancellationToken cancellationToken)
+	{
+		await RequestMicrophonePermission();
+
+		var microphonePermissionStatus = await Permissions.CheckStatusAsync<Permissions.Microphone>();
+		if (microphonePermissionStatus is not PermissionStatus.Granted)
+		{
+			throw new PermissionException("Microphone Permission Not Granted");
+		}
+
+		if (Connectivity.NetworkAccess is NetworkAccess.Internet)
+		{
+			return await ListenOnline(culture, recognitionResult, cancellationToken);
```

Comment:
I'm ok keeping both.

I was actually wondering if we should revert to Offline mode when Online fails (likely because the device is online but hasn't enabled the Voice Recording permission)?

Reply:
let's keep it as it is. we can allow users to choose the mode in the future, if necessary.